javatests/arcs/android/common/resurrection/ResurrectionRequestTest.kt

```diff
@@ -163,4 +163,62 @@ class ResurrectionRequestTest {
         }
         assertThat(ResurrectionRequest.createFromIntent(intent)).isNull()
     }
+
+    @Test
+    fun componentNameFromUnrequestIntent_returnsNull_ifWrongAction() {
+        val intent = Intent().apply {
+            action = "Not what we were looking for"
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_PACKAGE_NAME, "com.google.test")
+            putExtra(ResurrectionRequest.EXTRA_REGISTRATION_CLASS_NAME, "MyService")
```

Comment:
Same suggestion here as in the last PR. These could be clearer and less boilerplatey by using a constant and modifying only the interesting fields:

```
INTENT.clone().setAction(...)
```

(or `removeExtra`, etc.)